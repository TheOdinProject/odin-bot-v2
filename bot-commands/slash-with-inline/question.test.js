const question = require('./question-legacy');

describe('!question', () => {
  describe('regex', () => {
    it.each([
      ['!question'],
      [' !question'],
      ['!question @odin-bot'],
      ['@odin-bot !question'],
    ])('correct strings trigger the callback', (string) => {
      expect(question.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['!quest'],
      ['question'],
      ['!questio'],
      ['!questions'],
      ['```function("!question", () => {}```'],
      ['!questiona'],
      [''],
      [' '],
      [' !'],
      ['@odin-bot ! question'],
      ['!ques&tion'],
      ['!quest^ion'],
      ['!question!'],
      ['@odin-bot! question'],
      ['https:!!question.com'],
    ])("'%s' does not trigger the callback", (string) => {
      expect(question.regex.test(string)).toBeFalsy();
    });

    it.each([
      ['Check this out! !question'],
      ["Don't worry about !question"],
      ['Hey @odin-bot, !question'],
      ['!@odin-bot ^ !me !question !tests$*'],
    ])("'%s' - command can be anywhere in the string", (string) => {
      expect(question.regex.test(string)).toBeTruthy();
    });

    it.each([
      ['@user!question'],
      ["it's about!question"],
      ['!questionisanillusion'],
      ['!question!'],
      ['!question*'],
      ['!question...'],
    ])(
      "'%s' - command should be its own word!group - no leading or trailing characters",
      (string) => {
        expect(question.regex.test(string)).toBeFalsy();
      },
    );
  });

  describe('callback', () => {
    it('returns correct output', () => {
      expect(question.cb()).toMatchSnapshot();
    });
  });
});
