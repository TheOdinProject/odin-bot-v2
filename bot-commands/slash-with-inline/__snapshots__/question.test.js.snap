// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`!question callback returns correct output 1`] = `
{
  "embeds": [
    {
      "color": 13407555,
      "description": "
Asking context-rich questions makes it easy to receive help, and makes it easy for others to help you quickly! Great engineers ask great questions, and the prompt below is an invitation to improve your skills and set yourself up for success in the workplace.

**Project/Exercise:**
**Lesson link:**
**Code:** [code sandbox like replit or codepen]
**Issue/Problem:** [screenshots if applicable]
**What I expected:**
**What I've tried:**

For more information, read our [guide on honing your question-asking skills](<https://www.theodinproject.com/guides/community/how_to_ask>).
        ",
      "title": "Asking Great Questions",
    },
  ],
}
`;
